src/core/psxmem.cc

```diff
@@ -264,183 +262,154 @@ void PCSX::Memory::shutdown() {
     free(m_writeLUT);
 }
 
-uint8_t PCSX::Memory::read8(uint32_t mem) {
-    char *p;
-    uint32_t t;
+uint8_t PCSX::Memory::read8(uint32_t address) {
+    PCSX::g_emulator->m_cpu->m_regs.cycle += 1;
+    const uint32_t page = address >> 16;
+    const auto pointer = (uint8_t *)m_readLUT[page];
 
-    if (!PCSX::g_emulator->config().MemHack) {
-        PCSX::g_emulator->m_cpu->m_regs.cycle += 1;
-    }
-
-    t = mem >> 16;
-    if (t == 0x1f80 || t == 0x9f80 || t == 0xbf80) {
-        if ((mem & 0xffff) < 0x400)
-            return psxHu8(mem);
-        else
-            return PCSX::g_emulator->m_hw->read8(mem);
+    if (pointer != nullptr) {
+        const uint32_t offset = address & 0xffff;
+        return *(pointer + offset);
     } else {
-        p = (char *)(m_readLUT[t]);
-        if (p != NULL) {
-            return *(uint8_t *)(p + (mem & 0xffff));
+        if (page == 0x1f80 || page == 0x9f80 || page == 0xbf80) {
```

Comment:
Why not masking here? (or even earlier) Wouldn't it be faster than doing the `||` ?